nilearn/experimental/surface/maskers.py

```diff
@@ -247,8 +308,114 @@ def inverse_transform(self, masked_img: np.ndarray) -> SurfaceImage:
             data[part_name][..., mask] = masked_img[..., start:stop]
         return SurfaceImage(mesh=self.mask_img_.mesh, data=data)
 
+    def generate_report(self):
+        """Generate a report."""
+        try:
+            from nilearn.reporting.html_report import generate_report
+        except ImportError:
+            with warnings.catch_warnings():
+                mpl_unavail_msg = (
+                    "Matplotlib is not imported! "
+                    "No reports will be generated."
+                )
+                warnings.filterwarnings("always", message=mpl_unavail_msg)
+                warnings.warn(category=ImportWarning, message=mpl_unavail_msg)
+                return [None]
```

Comment:
Note that there is inconsistent behavior amongst the other maskers in nilearn regarding their `generate_report` when matplolib is missing.

NiftiMasker throws an error

```python
>>> from nilearn.maskers import NiftiMasker
>>> masker = NiftiMasker()
>>> masker.generate_report()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/remi/github/nilearn/nilearn/nilearn/maskers/nifti_masker.py", line 310, in generate_report
    from nilearn.reporting.html_report import generate_report
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/__init__.py", line 9, in <module>
    from nilearn.reporting.glm_reporter import make_glm_report
  File "/home/remi/github/nilearn/nilearn/nilearn/reporting/glm_reporter.py", line 22, in <module>
    from matplotlib import pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'
```

NiftiSpheresMasker throws an warning

```python
>>> from nilearn.maskers import NiftiSpheresMasker
>>> masker = NiftiSpheresMasker(seeds=(0,0,0))
>>> masker.generate_report()
/home/remi/github/nilearn/nilearn/nilearn/maskers/nifti_spheres_masker.py:409: ImportWarning: Matplotlib is not imported! No reports will be generated.
  warnings.warn(category=ImportWarning, message=mpl_unavail_msg)
[None]
```

Noticed this as the correct warnings in the maskers code are not tested.